cmd/app/link.go

```diff
@@ -156,10 +156,12 @@ func LinkExistingApp(ctx context.Context, clients *shared.ClientFactory, app *ty
 		}
 	}
 
-	// Confirm to update manifest source to remote
+	// Confirm to update manifest source to remote.
+	// - Update the manifest source to remote when its a GBP project with a local manifest.
+	// - Do not update manifest source for ROSI projects, because they can only be local manifests.
 	manifestSource, err := clients.Config.ProjectConfig.GetManifestSource(ctx)
 
-	if !manifestSource.Equals(config.MANIFEST_SOURCE_REMOTE) || err != nil {
+	if err != nil || (!manifestSource.Equals(config.MANIFEST_SOURCE_REMOTE) && cmdutil.IsSlackHostedProject(ctx, clients) != nil) {
```

Comment:
note: This is the main change. We only want to display the manifest source update prompt when the manifest is not already remote AND it's not a ROSI app.

Reply:
You and I were both low on brain power. I originally separated it to a variable in order to make sense of the conditional. Then I combined it afterwards. I'm happy to separate it out and I may do the same with the remote manifest conditional because the `!` and `.Equals` take a moment to make sense.

Reply:
Commt 30e69a7 takes your suggestion to heart and it's so much better!

```golang
isManifestSourceRemote := manifestSource.Equals(config.MANIFEST_SOURCE_REMOTE)
isSlackHostedProject := cmdutil.IsSlackHostedProject(ctx, clients) == nil

if err != nil || (!isManifestSourceRemote && !isSlackHostedProject) {
```